Neos.Neos/Resources/Public/JavaScript/Helper/createCollectionByPath.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * The function can be used to create objects or change values by path.
+ * So if you give an empty collection like {} with a path like a.b.c
+ * and a value, you get and object {a: {b: {c: 'value' } } } as response.
+ *
+ * You can also pass thru the collection  {a: {b: {c: 'value' } } } and change
+ * a.b.c to newValue. So you get  {a: {b: {c: 'newValue' } } }.
+ *
+ * If you use numerals as key we expect that the value should be an array
+ * instead of an object.
+ *
+ * @param {object} collection
+ * @param {string} path
+ * @param {mixed} value
+ * @return {object}
+ */
+const createCollectionByPath = (
+  collection: any,
+  path: string,
+  value: any
+) => {
+  collection = typeof collection === "object" ? collection : {};
+  const keys = Array.isArray(path) ? path : path.split(".");
+  let currentStep: object = collection;
+  for (let i = 0; i < keys.length - 1; i++) {
+    const key = keys[i];
+    if (
+      // @ts-ignore
```

Comment:
What's with all the ignores?

Reply:
That was the lazy solution to get little TS support :/